sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldBeInitializedCorrectlyBase.cs

```diff
@@ -0,0 +1,210 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class ObjectsShouldBeInitializedCorrectlyBase : SonarDiagnosticAnalyzer
+    {
+        protected abstract DiagnosticDescriptor Rule { get; }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        /// <summary>
+        /// Gets the KnownType representing the type on which instances the rule will raise issues on.
+        /// </summary>
+        internal abstract KnownType TrackedType { get; }
```

Comment:
Wouldn't it be better to merge the 2 properties (`TrackedType` and `TrackedPropertyName`) as only one using `MethodSignature` type?

Reply:
Perhaps, but it is a cosmetic change imo and not worth the time